src/goto-instrument/code_contracts.cpp

```diff
@@ -241,6 +244,88 @@ void code_contractst::add_quantified_variable(
   }
 }
 
+void code_contractst::replace_old_parameter(
+  exprt &expr,
+  std::map<exprt, exprt> &parameter2history,
+  source_locationt location,
+  const irep_idt &function,
+  const irep_idt &mode,
+  goto_programt &history)
+{
+  for(auto &op : expr.operands())
+  {
+    replace_old_parameter(
+      op, parameter2history, location, function, mode, history);
+  }
+
+  if(expr.id() == ID_old)
+  {
+    DATA_INVARIANT(
+      expr.operands().size() == 1, CPROVER_PREFIX "old must have one operand");
+
+    const auto &parameter = to_old_expr(expr).parameter();
+
+    // TODO: generalize below
+    PRECONDITION(parameter.id() == ID_dereference);
+
```

Comment:
`PRECONDITION` isn't meant for restrictions on user-input -- the idea is that these are provable.
Throw an exception whenever user input doesn't satisfy the restrictions.